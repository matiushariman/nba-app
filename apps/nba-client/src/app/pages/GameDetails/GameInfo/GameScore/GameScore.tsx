import TeamScore from './TeamScore';
import TeamLogo from '../../../../components/TeamLogo';
import GameStatusText from '../../../../components/GameStatusText';
import { selectGame } from '../../../../hooks/useGetGame';

const GameScore = () => {
  const game = selectGame();

  if (!game) {
    return null;
  }

  const { awayTeam, homeTeam, gameStatusText } = game;

  return (
    <div className="flex w-[370px] items-center" aria-label="game score">
      <TeamLogo
        teamId={awayTeam.teamId}
        teamName={awayTeam.teamName}
        height={52}
        width={52}
      />
      <TeamScore score={awayTeam.score} />
      <div className="flex-grow flex justify-around">
        <GameStatusText gameStatusText={gameStatusText} />
      </div>
      <TeamScore score={homeTeam.score} />
      <TeamLogo
        teamId={homeTeam.teamId}
        teamName={homeTeam.teamName}
        height={52}
        width={52}
      />
    </div>
  );
};

export default GameScore;
