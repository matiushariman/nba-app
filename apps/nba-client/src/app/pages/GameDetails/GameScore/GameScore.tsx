import TeamScore from './TeamScore';
import TeamLogo from '../../../components/TeamLogo';
import GameStatusText from '../../../components/GameStatusText';
import { selectGameScore } from '../../../hooks/useGetBoxscore';

const GameScore = () => {
  const game = selectGameScore();

  if (!game) {
    return null;
  }

  const { awayTeam, homeTeam, gameStatusText } = game;

  return (
    <div className="min-h-[64px] bg-white shadow flex items-center">
      <div className="container mx-auto flex justify-around md:justify-start divide-x">
        <div className="flex w-[370px] items-center" aria-label="game score">
          <TeamLogo
            teamId={awayTeam.teamId}
            teamName={awayTeam.teamName}
            height={52}
            width={52}
          />
          <TeamScore score={awayTeam.score} />
          <div className="flex-grow flex justify-around">
            <GameStatusText gameStatusText={gameStatusText} />
          </div>
          <TeamScore score={homeTeam.score} />
          <TeamLogo
            teamId={homeTeam.teamId}
            teamName={homeTeam.teamName}
            height={52}
            width={52}
          />
        </div>
      </div>
    </div>
  );
};

export default GameScore;
