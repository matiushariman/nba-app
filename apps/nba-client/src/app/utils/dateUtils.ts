import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
