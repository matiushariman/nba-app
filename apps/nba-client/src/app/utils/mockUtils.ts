export const shouldUseMock = process.env['NODE_ENV'] === 'development';
