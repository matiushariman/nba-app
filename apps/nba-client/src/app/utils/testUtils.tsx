import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
