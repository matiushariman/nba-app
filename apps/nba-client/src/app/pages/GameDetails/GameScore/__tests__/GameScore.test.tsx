import GameScore from '..';
import { render } from '../../../../utils/testUtils';

jest.mock('../../../hooks/useGetBoxscore', () => ({
  selectGameScore: jest.fn().mockReturnValueOnce({
    awayTeam: {
      teamId: '1',
      teamName: 'Lakers',
      score: '92',
    },
    homeTeam: {
      teamId: '2',
      teamName: 'Celtics',
      score: '131',
    },
    gameStatusText: 'FINAL',
  }),
}));

describe('pages/GameDetails/GameScore', () => {
  it('should render GameScore that matches snapshot', () => {
    const { baseElement } = render(<GameScore />);

    expect(baseElement).toMatchSnapshot();
  });
});
