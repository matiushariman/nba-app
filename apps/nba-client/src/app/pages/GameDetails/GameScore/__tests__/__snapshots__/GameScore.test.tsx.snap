// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/GameScore should render GameScore that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="min-h-[64px] bg-white shadow flex items-center"
    >
      <div
        class="container mx-auto flex justify-around md:justify-start divide-x"
      >
        <div
          aria-label="game score"
          class="flex w-[370px] items-center"
        >
          <img
            alt="Lakers logo"
            height="52"
            loading="lazy"
            src="https://cdn.nba.com/logos/nba/1/global/L/logo.svg"
            width="52"
          />
          <div
            class="w-[72px]"
          >
            <p
              class="text-3xl font-bold text-center"
            >
              92
            </p>
          </div>
          <div
            class="flex-grow flex justify-around"
          >
            <p
              class="text-sm uppercase font-bold"
            >
              FINAL
            </p>
          </div>
          <div
            class="w-[72px]"
          >
            <p
              class="text-3xl font-bold text-center"
            >
              131
            </p>
          </div>
          <img
            alt="Celtics logo"
            height="52"
            loading="lazy"
            src="https://cdn.nba.com/logos/nba/2/global/L/logo.svg"
            width="52"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
